lib/ui.lib.php

```diff
@@ -713,7 +713,7 @@ function toggle_visible($element)
 function display_notification($message, $timeout = 5000)
 {
     echo "<script type='text/javascript'>";
-    echo "displayNotification('" . addslashes(json_encode($message)) . "', " . $timeout . ");";
+    echo "displayNotification('" . addslashes(json_encode($message, JSON_UNESCAPED_UNICODE)) . "', " . $timeout . ");";
```

Comment:
From the JSON RFC:

> 2.5.  Strings
> 
>    The representation of strings is similar to conventions used in the C
>    family of programming languages.  A string begins and ends with
>    quotation marks.  All Unicode characters may be placed within the
>    quotation marks except for the characters that must be escaped:
>    quotation mark, reverse solidus, and the control characters (U+0000
>    through U+001F).
> 
>    Any character may be escaped.  If the character is in the Basic
>    Multilingual Plane (U+0000 through U+FFFF), then it may be
>    represented as a six-character sequence: a reverse solidus, followed
>    by the lowercase letter u, followed by four hexadecimal digits that
>    encode the character's code point.  The hexadecimal letters A though
>    F can be upper or lowercase.  So, for example, a string containing
>    only a single reverse solidus character may be represented as
>    "\u005C".

So just go for whichever is easier. Not escaping it is already implemented, let's stick with it?